src/core/parser/index.ts

```
@@ -0,0 +1,310 @@
+
+      // Validate and parse event
+      const validationResult = SessionEventSchema.safeParse(data);
+      if (!validationResult.success) {
+        continue;
+      }
+
+      const event = validationResult.data;
+      const eventTime = new Date(event.timestamp);
+
+      // Apply time filtering if provided
+      if (timeFilter && timeFilter.startTime && timeFilter.endTime) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)